src/client.js

```diff
@@ -3078,6 +3087,28 @@ MatrixClient.prototype.stopClient = function() {
     global.clearTimeout(this._checkTurnServersTimeoutID);
 };
 
+/*
+ * Query the server to see if it support members lazy loading
+ * @return {Promise<boolean>} true if server supports lazy loading
+ */
+MatrixClient.prototype.doesServerSupportLazyLoading = async function() {
+    if (this._serverSupportsLazyLoading === null) {
```

Comment:
We should probably do this somewhere common and/or do it once & cache it for whatever other features we want to detect down the line. Probably just as good to do this when the time comes though.

Reply:
I did it like this because it won't always be called from the same point. It's used from two places with the currently proposed PRs:
- in `MatrixClient::startClient` only if the `lazyLoadMembers` flag is enabled to make sure the server supports it.
- from `UserSettings` to confirm it is supported before enabling it from the labs checkbox.

So I didn't see a point in the code to always call it unconditionally from there and then use that cached value from then on.